examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,247 @@
+end
+nothing # hide
+
+# Choose the two fields to be output to be the total and perturbation vorticity.
+
+outputs = (ω_total = ω_field, ω_pert = ω_pert)
+
+# Build the `output_writer` for the two-dimensional fields to be output.
+# Output every `t = 1.0`.
+
+simulation.output_writers[:fields] =
+    NetCDFOutputWriter(
+        model,
+        (ω = ω_field, ωp = ω_pert),
+        filepath=joinpath(@__DIR__, "Bickley_jet_shallow_water.nc"),
+        schedule=TimeInterval(1.0),
+        mode = "c")
+
+# Build the `output_writer` for the growth rate, which is a scalar field.
+# Output every time step.
+#
+
+simulation.output_writers[:growth] =
+    NetCDFOutputWriter(
+        model,
+        (growth_rate = growth_rate,),
+        filepath="growth_rate_shallow_water.nc",
+        schedule=IterationInterval(1),
+        dimensions=(growth_rate=(),),
+        mode = "c")
+
+run!(simulation)
+
+# ## Visiualize the results
+#
+
+using NCDatasets, Plots, IJulia
+nothing # hide
+
+# Define the coordinates for plotting
+
+xf = xnodes(ω_field)
+yf = ynodes(ω_field)
+nothing # hide
+
+# Define keyword arguments for plotting the contours
+
+kwargs = (
+         xlabel = "x",
+         ylabel = "y",
+           fill = true,
+         levels = 20,
+      linewidth = 0,
+          color = :balance,
+       colorbar = true,
+           ylim = (0, Ly),
+           xlim = (0, Lx)
+)
+nothing # hide
+
+# Read in the `output_writer` for the two-dimensional fields
+# and then create an animation showing both the total and perturbation
+# vorticities.
+
+ds = NCDataset(simulation.output_writers[:fields].filepath, "r")
+
+iterations = keys(ds["time"])
+
+anim = @animate for (iter, t) in enumerate(ds["time"])
+        ω = ds["ω"][:,:,1,iter]
+    ωp = ds["ωp"][:,:,1,iter]
+
+     ω_max = maximum(abs, ω)
+    ωp_max = maximum(abs, ωp)
+
+     plot_ω = contour(xf, yf, ω',  clim=(-ω_max,  ω_max),  title=@sprintf("Total ω at t = %.3f", t); kwargs...)
+    plot_ωp = contour(xf, yf, ωp', clim=(-ωp_max, ωp_max), title=@sprintf("Perturbation ω at t = %.3f", t); kwargs...)
+
+    plot(plot_ω, plot_ωp, layout = (1,2), size=(1200, 500))
+
+    print("At t = ", t, " maximum of ωp = ", maximum(abs, ωp), "\n")
+end
+
+close(ds)
+
+mp4(anim, "Bickley_Jet_ShallowWater.mp4", fps=15)
+
+# Read in the `output_writer` for the scalar field.
+
+ds2 = NCDataset(simulation.output_writers[:growth].filepath, "r")
+
+iterations = keys(ds2["time"])
+
+t = ds2["time"][:]
+σ = ds2["growth_rate"][:]
+
+close(ds2)
+
+# Import `Polynomials` to be able to use `best_fit`.
+# Compute the best fit slope and save the figure.
+
+using Polynomials
+
+I = 60:70
+#I = 6000:7000
+best_fit = fit((t[I]), log.(σ[I]), 1)
+poly = 2 .* exp.(best_fit[0] .+ best_fit[1]*t[I])
+
+plt = plot(t[1000:end], log.(σ[1000:end]), lw=4, label="sigma", 
+        xlabel="time", ylabel="log(v)", title="growth rate", legend=:bottomright)
+plot!(plt, t[I], log.(poly), lw=4, label="best fit")
+
+print("Best slope = ", best_fit[1])
```

Comment:
In the similar spirit as the remark on `k = 0.95`, is this part necessary? What happens here is that you fit a polynomial to find the exponential growth. But there is no comparison of the numerical growth rate from the one predicted by theory, or is there? I think it'll be more useful if we say "for the parameters selected the predicted growth rate is ...." and then simply plot the line that corresponds to that growth rate for comparison on top of the energy plot.

Reply:
Thanks to your previous comment, I modified the introduction and stated that the growth rate predicted by theory (not showing any details) is \sigma \approx 0.14.  I have changed the final part to be the following.  What do you think?

```
# We can compute the slope of the curve on a log scale, which approximates the growth rate
# of the simulation.  This should be close to the theoretical prediction.

print("Growth rate in the simulation is approximated to be ", best_fit[1], 
      ", which is close to the theoretical value of 0.14.")
```